### PR TITLE
 use kubectl describe to implement karmadactl describe

### PR DESCRIPTION
**Pull request by wlp1153468871**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:
use kubectl describe to implement karmadactl describe in order to improve code reuse

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by lonelyCZ** [comment]
Could you please help to review it? @carlory

File: pkg/karmadactl/describe.go [hunk 1]
```diff
@@ -2,263 +2,115 @@ package karmadactl
...
+	kubectldescribe "k8s.io/kubectl/pkg/cmd/describe"
 	cmdutil "k8s.io/kubectl/pkg/cmd/util"
 	"k8s.io/kubectl/pkg/describe"
-	"k8s.io/kubectl/pkg/util/templates"
 
 	"github.com/karmada-io/karmada/pkg/karmadactl/options"
 )
 
-var (
-	describeExample = templates.Examples(`
```
Comment (carlory) [inline pkg/karmadactl/describe.go:22]: Thanks for your contribution  @wlp1153468871. Only a small sugguestion, others are nice.

keeping this part of the code, and use it like the following,

```go
Example:               fmt.Sprintf(describeExample, parentCommand),
```

Comment (wlp1153468871) [reply pkg/karmadactl/describe.go]: Done

File: pkg/karmadactl/describe.go [hunk 2]
```diff
@@ -71,194 +69,43 @@ func NewCmdDescribe(karmadaConfig KarmadaConfig, parentCommand string) *cobra.Co
...
 
 // Complete ensures that options are valid and marshals them if necessary
-func (o *CommandDescribeOptions) Complete(karmadaConfig KarmadaConfig, args []string) error {
-	var err error
-
-	if len(o.Cluster) == 0 {
-		return fmt.Errorf("must specify a cluster")
-	}
-
+func (o *DescribeOptions) Complete(karmadaConfig KarmadaConfig, cmd *cobra.Command, args []string) error {
```
Comment (carlory) [inline pkg/karmadactl/describe.go:94]: please add validation for `--cluster`

```go
	if len(o.Cluster) == 0 {
		return fmt.Errorf("must specify a cluster")
	}

```

Comment (wlp1153468871) [reply pkg/karmadactl/describe.go:94]: Done @carlory

**Comment by carlory** [comment]
Look great! Thanks @wlp1153468871 

/lgtm

cc @lonelyCZ

File: pkg/karmadactl/describe.go [hunk 3]
```diff
@@ -39,17 +35,19 @@ var (
 
 // NewCmdDescribe new describe command.
 func NewCmdDescribe(karmadaConfig KarmadaConfig, parentCommand string) *cobra.Command {
-	ioStreams := genericclioptions.IOStreams{In: getIn, Out: getOut, ErrOut: getErr}
-	o := &CommandDescribeOptions{
-		FilenameOptions: &resource.FilenameOptions{},
-		DescriberSettings: &describe.DescriberSettings{
-			ShowEvents: true,
-			ChunkSize:  cmdutil.DefaultChunkSize,
+	streams := genericclioptions.IOStreams{In: getIn, Out: getOut, ErrOut: getErr}
```
Comment (lonelyCZ) [inline pkg/karmadactl/describe.go:38]: We can rebase this branch after #2164 merged.

**Review by lonelyCZ — COMMENTED** [review]
Thanks for your contribution! We can hold this for waiting #2164 merged.

/hold

**Comment by lonelyCZ** [comment]
Hi @wlp1153468871 , could you please rebase this branch and resolve the conflicts?

File: pkg/karmadactl/describe.go (hunk 3 above)
Comment (wlp1153468871) [reply pkg/karmadactl/describe.go:38]: Yes, I'll deal with it tonight @lonelyCZ

**Comment by wlp1153468871** [comment]
> Hi @wlp1153468871 , could you please rebase this branch and resolve the conflicts?

Update

**Comment by carlory** [comment]
/hold cancel

leave `approve` to @lonelyCZ

**Review by RainbowMango — APPROVED** [review]
/approve
Thanks.

**Pull request merged**